### PR TITLE
MES 1916 - finalise pass page

### PR DESCRIPTION
**Pull request by buzzub**
## Description and relevant Jira numbers
- Create the Pass Finalisation page. 
- Recording whether the provisional license was provided is covered by MES-1096.
- Currently there's no validation on the pass certificate number. If none requested, possibly should add example text by the field to ensure valid data is entered.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

Screenshot 2019-03-28 at 16 56 46

Screenshot 2019-03-28 at 17 01 27

### CODE REVIEW

File: src/pages/pass-finalisation/pass-finalisation.ts [hunk 1]
```diff
@@ -23,8 +55,59 @@ export class PassFinalisationPage extends BasePageComponent {
...
+        select(getApplicationReference),
+        select(getApplicationNumber),
+      ),
+      passCertificateNumber$: this.store$.pipe(
+        select(getCurrentTest),
+        select(getPassCompletion),
+        select(getPassCertificateNumber),
+      ),
+    };
+    this.inputSubscriptions = [
```
Comment (johnmorleybjss) [inline src/pages/pass-finalisation/pass-finalisation.ts]: Missing an unsubscribe for these subscriptions.

File: src/theme/sass-partials/_input.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
-.vehicle-reg-input {
+.vehicle-reg-input,
+.pass-cert-input {
```
Comment (buzzub) [inline src/theme/sass-partials/_input.scss]: @johnmorleybjss Should we make the input field a generic class?

Comment (johnmorleybjss) [reply src/theme/sass-partials/_input.scss]: I was thinking we should probably just make all text inputs take this style without the need for an additional class with an attribute value selector, do you agree?

File: src/pages/pass-finalisation/pass-finalisation.ts [hunk 3]
```diff
@@ -23,8 +55,63 @@ export class PassFinalisationPage extends BasePageComponent {
     super(platform, navCtrl, authenticationProvider);
   }
 
+  ngOnInit(): void {
+    this.pageState = {
+      candidateName$: this.store$.pipe(
+        select(getCurrentTest),
```
Comment (szabolcs-keresztes) [inline src/pages/pass-finalisation/pass-finalisation.ts]: There is a `getTests` selector introduced which you can find in the `testsReducer`. Using that before getting the current test is necessary.

**Pull request merged**